Sources/SwiftDocC/SourceRepository/SourceRepository.swift

```diff
@@ -0,0 +1,92 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+/// A remote repository that hosts source code.
+public struct SourceRepository {
+    /// The path at which the repository is cloned locally.
+    public var checkoutPath: String
+    
+    /// The base URL where the service hosts the repository's contents.
+    public var sourceServiceBaseURL: URL
+    
+    /// A function that formats a line number to be included in a URL.
+    public var formatLineNumber: (Int) -> String
+    
+    /// Creates a source code repository.
+    /// - Parameters:
+    ///   - checkoutPath: The path at which the repository is checked out locally.
```

Comment:
Should we be more specific here? This needs to be the location where the repository was checked out when the given symbol graph was produced, right? Not just any local checkout.